fury/actors/effect_manager.py

```diff
@@ -0,0 +1,401 @@
+        if not isinstance(sigmas, np.ndarray):
+            sigmas = np.array(sigmas)
+        if sigmas.shape[0] != 1 and sigmas.shape[0] != points.shape[0]:
+            raise IndexError("sigmas size must be one or points size.")
+        if np.min(sigmas) <= 0:
+            raise ValueError("sigmas can't have zero or negative values.")
+
+        varying_dec = """
+        varying float out_sigma;
+        varying float out_scale;
+        """
+
+        kde_dec = import_fury_shader(
+            os.path.join("utils", f"{kernel.lower()}_distribution.glsl"))
+
+        kde_impl = """
+        float current_kde = kde(normalizedVertexMCVSOutput*out_scale, out_sigma);
+        color = vec3(current_kde);
+        fragOutput0 = vec4(color, 1.0);
+        """
+
+        kde_vs_dec = """
+        in float in_sigma;
+        varying float out_sigma;
+
+        in float in_scale;
+        varying float out_scale;
+        """
+
+        kde_vs_impl = """
+        out_sigma = in_sigma;
+        out_scale = in_scale;
+        """
+
+        tex_dec = import_fury_shader(os.path.join("effects", "color_mapping.glsl"))
+
+        tex_impl = """
+        // Turning screen coordinates to texture coordinates
+        vec2 res_factor = vec2(res.y/res.x, 1.0);
+        vec2 renorm_tex = res_factor*normalizedVertexMCVSOutput.xy*0.5 + 0.5;
+        float intensity = texture(screenTexture, renorm_tex).r;
+
+        if(intensity<=0.0){
+            discard;
+        }else{
+            vec4 final_color = color_mapping(intensity, colormapTexture);
+            fragOutput0 = vec4(final_color.rgb, u_opacity*final_color.a);
+        }
+        """
+
+        fs_dec = compose_shader([varying_dec, kde_dec])
+
+        """Scales parameter will be defined by the empirical rule:
+        1*sima radius = 68.27% of data inside the curve
+        2*sigma radius = 95.45% of data inside the curve
+        3*sigma radius = 99.73% of data inside the curve"""
+        scales = 2*3.0*np.copy(sigmas)
+
+        center_of_mass = np.average(points, axis=0)
+        bill = billboard(
+            points,
+            (0.0,
+             0.0,
+             1.0),
+            scales=scales,
+            fs_dec=fs_dec,
+            fs_impl=kde_impl,
+            vs_dec=kde_vs_dec,
+            vs_impl=kde_vs_impl)
+
+        # Blending and uniforms setup
+        window = self.off_manager.window
+
+        shader_apply_effects(window, bill, gl_disable_depth)
+        shader_apply_effects(window, bill, gl_set_additive_blending)
+        attribute_to_actor(bill, np.repeat(sigmas, 4), "in_sigma")
+        attribute_to_actor(bill, np.repeat(scales, 4), "in_scale")
+
+        if self._n_active_effects > 0:
+            self.off_manager.scene.GetActors().GetLastActor().SetVisibility(False)
+        self.off_manager.scene.add(bill)
+
+        bill_bounds = bill.GetBounds()
+        max_sigma = 2*4.0*np.max(sigmas)
+
+        actor_scales = np.array([[bill_bounds[1] - bill_bounds[0] +
```

Comment:
May I know how are you calculating the correct scaling?

Reply:
Sure. The actor_scales is a dynamic variable that calculates the general scale of the `textured_billboard` depending on some parameters:
* `bill_bounds` : The bounds of the offscreen rendered billboards
* `center_of_mass` : The average of all of the points positions. This accounts for the possible displacement of the points
* `max_sigma` : The maximum value that includes all of the non-zero values of the KDE calculation. It relates to the empirical rule of the gaussian disstribution, that says that $3\sigma$ includes 99.73% of the values, so $4\sigma$ should include the remaining ones.

All of them are them summed together to calculate a possible size so the billboard would not cut the renderization. This is like this for now, but I have identified that I may need to change the scaling to a way that it occupies the whole screen, as it is affecting the real size of the points, so expect some change regarding that.